deepchecks/vision/vision_data/batch_wrapper.py

```diff
@@ -0,0 +1,179 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Contains code for BatchWrapper."""
+from typing import Any, Callable, Dict, List, Sequence, TypeVar, Union, cast
+
+import numpy as np
+import torch
+
+from deepchecks.vision.utils.image_functions import crop_image
+from deepchecks.vision.utils.vision_properties import PropertiesInputType, calc_vision_properties, validate_properties
+from deepchecks.vision.vision_data import VisionData
+from deepchecks.vision.vision_data.utils import BatchOutputFormat, sequence_to_numpy
+
+__all__ = ['BatchWrapper']
+
+
+class BatchWrapper:
+    """Represents dataset batch returned by the dataloader during iteration."""
+
+    def __init__(
+            self,
+            batch: BatchOutputFormat,
+            vision_data: VisionData,
+    ):
+        self._vision_data = vision_data
+        self._batch = batch
+        self._labels, self._predictions, self._images = None, None, None
+        self._embeddings, self._additional_data, = None, None
+        self._image_identifiers = batch.get('image_identifiers')
+        # if there are no image identifiers, use the number of  the image in loading process as identifier
+        if self._image_identifiers is None:
+            images_seen = self._vision_data.number_of_images_cached
+            self._image_identifiers = np.asarray(range(images_seen, images_seen + self.__len__()), dtype='str')
+
+        self._vision_properties_cache = dict.fromkeys(PropertiesInputType)
+
+    def _get_cropped_images(self):
+        imgs = []
+        for img, labels in zip(self.images, self.labels):
+            for label in labels:
+                label = label.cpu().detach().numpy()
+                bbox = label[1:]
+                # make sure image is not out of bounds
+                if round(bbox[2]) + min(round(bbox[0]), 0) <= 0 or round(bbox[3]) <= 0 + min(round(bbox[1]), 0):
+                    continue
+                imgs.append(crop_image(img, *bbox))
+        return imgs
+
+    def _get_relevant_data_for_properties(self, input_type: PropertiesInputType):
+        if input_type == PropertiesInputType.PARTIAL_IMAGES:
+            return self._get_cropped_images()
+        if input_type == PropertiesInputType.IMAGES:
+            return self._batch.get('images')
+        if input_type == PropertiesInputType.LABELS:
+            return self._batch.get('labels')
```

Comment:
Why do these instead of self.numpy_labels etc?

Reply:
True, on it